### PR TITLE
workflows: add remove-disabled-formulae workflow

### PR DESCRIPTION
**Pull request by Rylan12**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR adds a workflow that will automatically remove formulae that have a disable date of more than one year ago. See discussion in #66360. I haven't really tested the actual PR-creation part of this (although I've tested the old-formula-detection part locally), so if it seems okay I'd suggest merging and seeing what happens.

### CODE REVIEW

**Comment by dawidd6** [comment]
This is great and all, but...
- I would like to see this Action added to Homebrew/actions first
- An Action for pull request creation could be created from scratch for us, we have a bunch of workflows in Homebrew/brew that would benefit from it too
- Use `brew ruby` to run the script
- Add `disable_date` method to `formula.rb` in Homebrew/brew, as I see it isn't available now
- Sweep all formulae to get deprecate dates via `formula.rb` API

**Comment by Rylan12** [comment]
Good suggestions

> * I would like to see this Action added to Homebrew/actions first

Seems fine. I didn't start by adding it there because it felt too specific to the homebrew/core repo. No problem to add it there, though.

> * An Action for pull request creation could be created from scratch for us, we have a bunch of workflows in Homebrew/brew that would benefit from it too

Why reinvent the wheel? I don't see a reason to spend time and effort creating out own when one that does the job perfectly already exists. 

> * Use `brew ruby` to run the script

I see there's an action in the actions repo for this already ([`brew-script`](https://github.com/Homebrew/actions/tree/master/brew-script)). Can this be used by another action in Homebrew/actions or should I just model it after the existing `brew-script` action?

> * Add `disable_date` method to `formula.rb` in Homebrew/brew, as I see it isn't available now
> * Sweep all formulae to get deprecate dates via `formula.rb` API

Yeah, that's probably a better way to do it. I was trying to avoid making unnecessary changes but it's probably a good idea given that deprecation/disable dates are now required.

**Comment by dawidd6** [comment]
Can take `brew-script` and repurpose it.

About the wheel reinventing, I just think we should use only official Actions or the ones that are written by ourselves, but if others agree with you then fine.

**Comment by Rylan12** [comment]
> I just think we should use only official Actions or the ones that are written by ourselves, but if others agree with you then fine.

The sense I have is that we don't mind using third-party actions as long as they're pinned to a specific commit. For now, I'll continue to use the third-party action but if the consensus is that having our own is helpful (or someone else writes one for Homebrew) I have no problem adjusting.

In the meantime, I'll work on this some more tonight and see if I can get the other main points addressed.

**Comment by carlocab** [comment]
Re re-inventing the wheel:

It's not just about not duplicating the world that someone else has already done, but it's also about avoiding the duplication of the work that someone else is going to do. Unless Homebrew has special requirements from a create-a-pull-request action, it seems like starting from scratch also entails unnecessarily shouldering the maintenance burden of that action.

**Review by MikeMcQuaid — DISMISSED** [review]
> Unless Homebrew has special requirements from a create-a-pull-request action, it seems like starting from scratch also entails unnecessarily shouldering the maintenance burden of that action.

There's also the maintenance of "updating the pinned commit" and "monitoring the security" of relying on an upstream action.

I think for something as relatively simple as creating a pull request it'd be good to have our own action that we can use amongst the many other actions we have that do so.

I don't think this needs to block this PR, though.

**Comment by Rylan12** [comment]
I've opened a PR in the brew repo in https://github.com/Homebrew/brew/pull/10108 actions repo in https://github.com/Homebrew/actions/pull/133. Once those are handled (in that order) this should be ready again.

**Comment by carlocab** [comment]
Does this enable a final review before candidate disabled formulae are removed? Or does one need to be paying attention to workflow runs to tell whether a disabled formula has been removed?

I don't mind the lack of a final check; I just want to be clear about the consequences of this workflow for disabled formulae.

**Comment by Rylan12** [comment]
No, this will simply open a PR that removes the formulae. It will still need the usual maintainer approval and all that stuff.

**Comment by carlocab** [comment]
Ah, I was looking at your two other PRs which lead me to the impression that it would silently delete disabled formulae. I should've been looking at this one. 😅

File: .github/workflows/remove-disabled-formulae.yml
Comment (dawidd6) [inline .github/workflows/remove-disabled-formulae.yml:27]:
```suggestion
        uses: Homebrew/actions/remove-disabled-formulae@master
```

File: .github/workflows/remove-disabled-formulae.yml
Comment (dawidd6) [inline .github/workflows/remove-disabled-formulae.yml:30]:
```suggestion
        if: ${{ steps.remove_disabled.outputs.formulae-removed == 'true' }}
```

File: .github/workflows/remove-disabled-formulae.yml
Comment (Rylan12) [reply .github/workflows/remove-disabled-formulae.yml:27]: Oops, good catch 🤦

File: .github/workflows/remove-disabled-formulae.yml
Comment (Rylan12) [reply .github/workflows/remove-disabled-formulae.yml:30]: Thanks 👍

**Comment by Rylan12** [comment]
I tested this [here](https://github.com/Rylan12/homebrew-development/actions/runs/441368161) and it seems to work as expected. I'm going to merge and we'll see how it goes!

**Comment by Rylan12** [comment]
Success! 🎉🥳 

https://github.com/Homebrew/homebrew-core/pull/67591

**Pull request merged**
